includes/class-wc-admin-reports-interval.php

```diff
@@ -422,9 +461,6 @@ public static function next_year_start( $datetime, $reversed = false ) {
 	 * @return DateTime
 	 */
 	public static function iterate( $datetime, $time_interval, $reversed = false ) {
-		// $result_datetime           =
-		// $result_timestamp_adjusted = $result_datetime->format( 'U' ) - 1;
-		// $result_datetime->setTimestamp( $result_timestamp_adjusted );
 		return call_user_func( array( __CLASS__, "next_{$time_interval}_start" ), $datetime, $reversed );
```

Comment:
Since it looks like `$time_interval` comes from the `query` args, should we do any checking to be safe that it is a valid interval function before calling it?

Reply:
I think it's validated for REST queries via the controllers, e.g. 

https://github.com/woocommerce/wc-admin/blob/368c02e53086f9f3a21451e14b2b08ab664fbd15/includes/api/class-wc-admin-rest-reports-orders-stats-controller.php#L366-L379

However, if used directly as `WC_Admin_Reports_*_Query` or `WC_Admin_Reports_*_Data_Store`, then invalid value might result in fatals, so it's a valid concern. I think there are more args where this is true.

Reply:
Ah yeah, good point on REST handing validation.

It might be worth adding some protection to this and other args later, but definitely doesn't need to be a blocker. 👍